### PR TITLE
feat: update the Component Explorer to use web components and native HTML elements

### PR DESCRIPTION
**Pull request by janechu**
# Description

This updates the Component Explorer site to use web components and native HTML Elements. Previously this site had been used for the React component package `@microsoft/fast-components-react-msft`, now this is a showcase for the web components package `@microsoft/fast-components` package.

## Motivation & context

In preparation for launching the web component package `@microsoft/fast-components`, this site has also been updated to show examples of these components and native elements where necessary. This pull request has been issued against a feature branch from a series of PRs that have already been merged, namely:

- https://github.com/microsoft/fast-dna/pull/3026
- https://github.com/microsoft/fast-dna/pull/3040
- https://github.com/microsoft/fast-dna/pull/3117
- https://github.com/microsoft/fast-dna/pull/3141
- https://github.com/microsoft/fast-dna/pull/3178
- https://github.com/microsoft/fast-dna/pull/3191
- https://github.com/microsoft/fast-dna/pull/3209
- https://github.com/microsoft/fast-dna/pull/3235

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: sites/fast-component-explorer/app/index.tsx [hunk 1]
```diff
@@ -11,10 +11,9 @@ const root: HTMLElement = document.createElement("div");
...
 
-const DragDropApp: any = DragDropContext(HTML5Backend)(App);
-
-function render(): void {
-    ReactDOM.render(<DragDropApp />, root);
-}
-
-render();
+ReactDOM.render(
+    <DndProvider backend={HTML5Backend}>
```
Comment (EisenbergEffect) [inline sites/fast-component-explorer/app/index.tsx:15]: Every time I see this I can't help but think "Dungeons and Dragons" 🤣

**Comment by janechu** [comment]
@EisenbergEffect you would need to do a yarn install first, but yep, that's all.

File: sites/fast-component-explorer/app/preview.style.ts [hunk 2]
```diff
@@ -15,20 +17,31 @@ const style: ComponentStyles<PreviewClassNameContract, DesignSystem> = {
...
     },
     preview: {
-        height: "100%",
         boxSizing: "border-box",
     },
+    preview__transparent: {
+        background:
+            "linear-gradient(45deg, rgba(0, 0, 0, 0.0980392) 25%, transparent 25%, transparent 75%, rgba(0, 0, 0, 0.0980392) 75%, rgba(0, 0, 0, 0.0980392) 0), linear-gradient(45deg, rgba(0, 0, 0, 0.0980392) 25%, transparent 25%, transparent 75%, rgba(0, 0, 0, 0.0980392) 75%, rgba(0, 0, 0, 0.0980392) 0), white",
+        "background-repeat": "repeat, repeat",
+        "background-position": "0px 0, 5px 5px",
```
Comment (eljefe223) [inline sites/fast-component-explorer/app/preview.style.ts:30]: `0px` can just be `0`

File: sites/fast-component-explorer/app/fast-components/configs/component-configs.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import fastAnchorConfig from "./fast-anchor";
+export { fastAnchorConfig };
+
```
Comment (nicholasrice) [inline sites/fast-component-explorer/app/fast-components/configs/component-configs.ts:3]: for future reference you can just do:

```js
export { fastAnchorConfig } from "./fast-anchor";
```

**Pull request merged**